### PR TITLE
[ethtrader-gov-weight] strategy for r/ethtrader dao governance

### PR DESCRIPTION
**Pull request by kohrts**
strategy fetches governance weights from a json that we manually update every 4 weeks during donut & contrib distributions to contributing DAO members on the r/ethtrader subreddit

### CODE REVIEW

File: src/strategies/index.ts [hunk 1]
```diff
@@ -528,6 +529,7 @@ const strategies = {
   starsharks,
   'printer-financial': printerFinancial,
   'ethercats-founders-series': ethercatsFoundersSeries,
+  'ethtrader-gov-weight': ethtraderGovWeight
```
Comment (ChaituVR) [inline src/strategies/index.ts:532]:
```suggestion
  'ethtrader-gov-weight': ethtraderGovWeight,
```

File: src/strategies/ethtrader-gov-weight/index.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import fetch from 'cross-fetch';
+
+export const author = 'kohrts';
+export const version = '0.1.0';
+
+export async function strategy(space, network, provider, addresses, options, snapshot) {
+  const ETHTRADER_USERS_URL = 'https://raw.githubusercontent.com/EthTrader/donut.distribution/main/docs/users.json'
```
Comment (ChaituVR) [inline src/strategies/ethtrader-gov-weight/index.ts:7]: Hi, @kohrts  like this owner of this repo can change the data any time and won't give us consistent results, We need this data from on-chain / IPFS so it won't change in the future, Can you think of any alternate way to get data based on block and network?

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/strategies/ethtrader-gov-weight/index.ts (hunk 2 above)
Comment (kohrts) [reply src/strategies/ethtrader-gov-weight/index.ts:7]: Yes, all this data does originate on-chain and IPFS, but it requires more than 5 calls to compile it all.

Comment (bonustrack) [reply src/strategies/ethtrader-gov-weight/index.ts:7]: Is it 5 calls in total or 5 calls per address? If it's in total that should work within the strategy. If you use external service to get voting power you need to take the "snapshot" in consideration and store voting power history in a way that it always answer with same power based on the snapshot

File: src/strategies/ethtrader-gov-weight/index.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  
+  const response = await fetch(ETHTRADER_USERS_URL, {
```
Comment (bonustrack) [inline src/strategies/ethtrader-gov-weight/index.ts:9]: "snapshot" should be used here, you don't just need voting power, you need voting power at a specific snapshot / time

File: src/strategies/ethtrader-gov-weight/index.ts (hunk 2 above)
Comment (kohrts) [reply src/strategies/ethtrader-gov-weight/index.ts:7]: thanks, i get it! that makes a lot of sense. i'll make those updates

**Pull request closed without merging**